alf/layers.py

```diff
@@ -3612,3 +3612,34 @@ def forward(self, input):
                 lambda x: x.float() if x.dtype.is_floating_point else x, input)
         with torch.cuda.amp.autocast(self._enabled):
             return self._net(input)
+
+
+class MultiHeadAttention(nn.Module):
+    """Multi-Head Attention Module."""
+
+    def __init__(self):
+        super().__init__()
+
+    def forward(self, query, key, value):
+        """Multi-Head attention computation based on the inputs.
+        Args:
+            query (Q): shape [B, head, 1, d]
+            key (K):  shape [B, head, N, d]
+            value (V): shape [B, head, N, d]
+            where B denotes the batch size, head denotes the number of heads,
+            N the number of entities, and d the feature dimension.
+        Return:
+            - the attended results computed as: softmax(QK^T/sqrt(d))V
+            - the attention weight, with the shape [B, head, 1, d]
```

Comment:
The shape for attention weight is [B, head, 1, N]?

Reply:
Fixed.